### PR TITLE
 Donor on boarding flow

### PR DESCRIPTION
**Pull request by JeanVegaD**
# What does this PR do?

#219 

- update login and signup with email
- add authentication with Facebook and Google 

# How this should be tested?

- make run
- go to http://localhost:3000
- try to login/signup with email, Facebook, or Google

### CODE REVIEW

File: webapp/src/components/LoginModal/LoginWithFacebook.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  },
+  facebookIcon: {
+    marginRight: 5
+  }
+}))
+
+const LoginWithFacebook = ({ onSubmit }) => {
+  const classes = useStyles()
+
+  const responseFacebook = (response) => {
```
Comment (JustinCast) [inline webapp/src/components/LoginModal/LoginWithFacebook.js:25]: `facebookResponse` ?

File: webapp/src/components/LoginModal/LoginWithFacebook.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    onSubmit(true, response.profile.email, response.profile.id)
+  }
+
+  const handleError = (error) => {
+    if (error) {
+      onSubmit(false, "", "")
+    }
+
```
Comment (JustinCast) [inline webapp/src/components/LoginModal/LoginWithFacebook.js:33]: remove the blank space and make the validation in one single line

File: webapp/src/components/LoginModal/LoginWithFacebook.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  return (
+    <>
+      <FacebookProvider appId="641671940102186">
```
Comment (JustinCast) [inline webapp/src/components/LoginModal/LoginWithFacebook.js:38]: What about if we load this from a config file?

File: webapp/src/gql/account.gql.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import gql from 'graphql-tag'
 
 export const CREATE_ACCOUNT_MUTATION = gql`
-  mutation($role: String!, $username: String!, $secret: String!) {
-    create_account(role: $role, username: $username, secret: $secret) {
+  mutation($role: String!, $email: String!,$name:String! $secret: String!) {
+    create_account(role: $role, email: $email, name:$name, secret: $secret) {
```
Comment (JustinCast) [inline webapp/src/gql/account.gql.js:5]: add white `space name:$name` to `name: $name`

File: webapp/src/routes/Signup/ValidateEmail.js [hunk 5]
```diff
@@ -48,6 +48,8 @@ const ValidateEmail = ({ isValid, loading, user, setField }) => {
         validateFormatEmail(user.email) && !isValid && loading
           ? 'This email already has an associated account'
           : !isValid
+            ? ''
```
Comment (JustinCast) [inline webapp/src/routes/Signup/ValidateEmail.js:51]: Please avoid using complex validations

File: webapp/src/routes/Signup/socialSingup/SignupWithFacebook.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+const SignupWithFacebook = ({ handlerSubmit }) => {
+  const classes = useStyles()
+
+  const responseFacebook = (response) => {
+    handlerSubmit(true, response.profile.email, response.profile.name, response.profile.id)
+  }
+
+  const handleError = (error) => {
+    if (error) {
+      handlerSubmit(false, "", "", "")
```
Comment (JustinCast) [inline webapp/src/routes/Signup/socialSingup/SignupWithFacebook.js:35]: Make the validation in one single line

File: webapp/src/routes/Signup/socialSingup/SignupWithFacebook.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  }
+
+  return (
+    <>
+      <FacebookProvider appId="641671940102186">
```
Comment (JustinCast) [inline webapp/src/routes/Signup/socialSingup/SignupWithFacebook.js:41]: Maybe is better to load this from a config file

File: webapp/src/routes/Signup/socialSingup/SignupWithGoogle.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+const SignupWithGoogle = ({ handlerSubmit }) => {
+  const classes = useStyles()
+
+  const responseGoogle = (response) => {
+    handlerSubmit(true, response.profileObj.email, response.profileObj.name, response.profileObj.googleId)
+  }
+
+  const handleError = (error) => {
+    if (error) {
```
Comment (JustinCast) [inline webapp/src/routes/Signup/socialSingup/SignupWithGoogle.js:35]: Make the validation in one single line

File: webapp/src/routes/Signup/socialSingup/SignupWithGoogle.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+      handlerSubmit(false, "", "", "")
+    }
+  }
+
+  return (
+    <GoogleLogin
+      clientId="865299157499-ht14599cb2vaqgnq0vhkihsedsfvcukf.apps.googleusercontent.com"
```
Comment (JustinCast) [inline webapp/src/routes/Signup/socialSingup/SignupWithGoogle.js:42]: Maybe is better to load this from a config file

**Review by JustinCast — CHANGES_REQUESTED** [review]

File: webapp/src/components/LoginModal/LoginWithFacebook.js [hunk 10]
```diff
@@ -22,23 +24,21 @@ const useStyles = makeStyles((theme) => ({
...
-  const responseFacebook = (response) => {
+  const facebookResponse = (response) => {
     onSubmit(true, response.profile.email, response.profile.id)
   }
 
   const handleError = (error) => {
-    if (error) {
-      onSubmit(false, "", "")
-    }
+    if (error) { onSubmit(false, "", "") }
```
Comment (JustinCast) [inline webapp/src/components/LoginModal/LoginWithFacebook.js:32]: When the validation has only one line in its body, you can remove curly braces. So, the result is `if (error) onSubmit(false, "", "")`

File: webapp/src/components/LoginModal/LoginWithGoogle.js [hunk 11]
```diff
@@ -24,19 +26,17 @@ const useStyles = makeStyles((theme) => ({
...
-  const responseGoogle = (response) => {
+  const googleResponse = (response) => {
     onSubmit(true, response.profileObj.email, response.profileObj.googleId)
   }
 
   const handleError = (error) => {
-    if (error) {
-      onSubmit(false, "", "")
-    }
+    if (error) { onSubmit(false, "", "") }
```
Comment (JustinCast) [inline webapp/src/components/LoginModal/LoginWithGoogle.js:34]: unnecessary curly braces here

**Review by JustinCast — CHANGES_REQUESTED** [review]

**Review by JustinCast — APPROVED** [review]

**Review by JustinCast — APPROVED** [review]

File: webapp/src/components/LoginModal/LoginWithFacebook.js [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+  const facebookResponse = (response) => {
+    onSubmit(true, response.profile.email, response.profile.id)
+  }
+
+  const handleError = (error) => {
+    if (error) onSubmit(false, "", "")
+  }
+
+  return (
+    <>
```
Comment (tetogomez) [inline webapp/src/components/LoginModal/LoginWithFacebook.js:36]: este `fragment` es necesario ?

File: webapp/src/routes/Signup/socialSingup/SignupWithFacebook.js [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+  const facebookResponse = (response) => {
+    handlerSubmit(true, response.profile.email, response.profile.name, response.profile.id)
+  }
+
+  const handleError = (error) => {
+    if (error) handlerSubmit(false, "", "", "")
+  }
+
+  return (
+    <>
```
Comment (tetogomez) [inline webapp/src/routes/Signup/socialSingup/SignupWithFacebook.js:40]: same here

**Review by tetogomez — COMMENTED** [review]
Please check the comments

**Review by murillojorge — CHANGES_REQUESTED** [review]
Please review unique ids on user table.

 FatalError: there is no unique constraint matching given keys for referenced table \"user\""

File: package.json [hunk 14]
```diff
@@ -0,0 +1,3 @@
+{
+  "dependencies": {}
```
Comment (xavier506) [inline package.json:2]: @JeanVegaD  do we need this file?

File: webapp/src/routes/Signup/Signup.js [hunk 15]
```diff
@@ -120,14 +128,53 @@ const Signup = () => {
...
+                role,
+                email,
+                name,
+                secret: hash
+              }
+            })
+          }
+        });
+
+      } else {
```
Comment (JustinCast) [inline webapp/src/routes/Signup/Signup.js:174]: Unnecessary curly braces here

File: webapp/src/routes/Signup/Signup.js [hunk 16]
```diff
@@ -120,14 +128,53 @@ const Signup = () => {
...
+
+      const { data } = await checkEmail({ email: email })
+
+      if (data.user.length === 0) {
+
+        const bcrypt = require('bcryptjs');
+        const saltRounds = 10;
+
+        bcrypt.hash(secret, saltRounds, function (err, hash) {
+          if (!err) {
```
Comment (JustinCast) [inline webapp/src/routes/Signup/Signup.js:162]: Unnecessary curly braces here

**Review by JustinCast — CHANGES_REQUESTED** [review]

**Review by AngeloCG97 — APPROVED** [review]

**Review by JustinCast — APPROVED** [review]

**Pull request merged**